specs/error_state/ErrorOutOfGasDynamicMemoryExpansion.md

```diff
@@ -0,0 +1,59 @@
+# ErrorOutOfGasDynamicMemoryExpansion state
+
+## Procedure
+
+Handle the corresponding out of gas errors for `CREATE`, `RETURN` and `REVERT` opcodes.
+
+### EVM behavior
+
+For the current `go-ethereum` code, the out of gas error may occur for `constant gas` or `dynamic gas`. The gas cost is calculated as:
+
+```
+gas_cost = constant_gas + dynamic_gas
+```
+
+The constant gas is different for `CREATE`, `RETURN` and `REVERT`.
+
+```
+if opcode == CREATE:
+    constant_gas = 32000
+else:
+    constant_gas = 0
+```
+
+They are also same for dynamic gas calculation. The dynamic gas is calculated as:
```

Comment:
Nitpick: I would remove the first sentence ("They are also same for dynamic gas calculation.") as it says "also same", but the sentence above says "is different".